packages/next/src/build/webpack-config.ts

```diff
@@ -1180,6 +1195,19 @@ export default async function getBaseWebpackConfig(
       webassemblyModuleFilename: 'static/wasm/[modulehash].wasm',
       hashFunction: 'xxhash64',
       hashDigestLength: 16,
+
+      ...(config.experimental.flyingShuttle
+        ? {
+            // ensure we only use contenthash as it's more deterministic
+            filename: isNodeOrEdgeCompilation
+              ? dev || isEdgeServer
+                ? `[name].js`
+                : `../[name].js`
+              : `static/chunks/${isDevFallback ? 'fallback/' : ''}[name]${
+                  dev ? '' : appDir ? '-[contenthash]' : '-[contenthash]'
```

Comment:
Seems like we don't need the extra condition since it's the same in both cases
```suggestion
                  dev ? '' : '-[contenthash]'
```